docs/src/design/lookups/logup.md

```diff
@@ -0,0 +1,65 @@
+# LogUp: multivariate lookups with logarithmic derivatives
+
+The description of LogUp can be found [here](https://eprint.iacr.org/2022/1530.pdf). In MidenVM, LogUp is used to implement efficient [communication buses](./main.md#communication-buses-in-miden-vm). 
+
+Using the LogUp construction instead of a simple [multiset check](./multiset.md) with running products reduces the computational effort for the prover and the verifier. Given two columns $a$ and $b$ in the main trace where $a$ contains duplicates and $b$ does not (i.e. $b$ is part of the lookup table), LogUp allows us to compute two logarithmic derivatives and check their equality.
+
+$$
+\sum_{i=0}^{l} \frac{1}{(\alpha - a_i)} = \sum_{i=0}^{n} \frac{m_i}{(\alpha - b_i)}
+$$
+
+In the above:
+- $l$ is the number of values in $a$
+- $n$ is the number of values in $b$
+- $m_i$ is the multiplicity of $b_i$, which is expected to match the number of times the value $b_i$ is duplicated in column $a$
+- $\alpha$ is a random value that is sent to the prover by the verifier after the prover commits to the execution trace of the program.
+
+Thus, instead of needing to compute running products, we are able to assert correct lookups by computing running sums.
```

Comment:
I believe there is a limitation on the value of multiplicity - but I don't remember what it is exactly. We should probably mention it here.

Reply:
I was thinking the multiplicity was limited by the size of the field, which I stated somewhere. Is there another limitation?

Reply:
It's either that or the size of the domain (or I wonder if having multiplicity bigger than the size of the domain causes some kind of security degradation).

Reply:
Both $l$ and $n$ should be smaller than $p$ the size of field, from the paper. The multiplicities should be smaller than the size of the set we are looking up into, that's my understanding from what I recall about the paper.

Reply:
I've updated this section, as well as at `docs/src/design/range.md` line 62,